add sprites and adaptive img

ThePen4iK opened the pull request.

ThePen4iK committed:
add sprites and adaptive img

WebnautsAcademy commented on source/index.html:
```diff
@@ -58,20 +50,23 @@
-              alt="main image"
-              class="start__img"
-            />
+            <picture>
```
В теге picture должны быть вложенные source с медиа условиями, если сайт адаптивный

WebnautsAcademy commented on source/index.html:
```diff
@@ -90,14 +85,16 @@ <h1 class="start__title title">
           <div class="about__wrapper-view">
             <img
               src="img/about-life.jpg"
+              srcset="./img/about-life@2x.jpg"
```
Тут после 2х.jpg  ещё 2х должно идти

WebnautsAcademy commented on source/index.html:
```diff
@@ -108,10 +105,95 @@ <h1 class="start__title title">
         </div>
       </section>
       <section class="main">
```
А почему у секции класс майн? А что у тега майн за класс тогда? Лучше указывать .main только <main>,  а другим секциям - любые другие названия

WebnautsAcademy commented on source/index.html:
```diff
@@ -121,60 +203,78 @@ <h1 class="start__title title">
+          <div class="project__logo">
+            <img
+              src="img/project-logo.png"
+              srcset="./img/project-logo@2x.png"
```
Проверьте все srcset у img и picture. Упущено условие 2х и в данном случае. И протестируйте в инструментах разработчика подключение ретиновых картинок

WebnautsAcademy commented on source/js/slider.js:
```diff
@@ -0,0 +1,10 @@
+var swiper = new Swiper(".mySwiper", {
```
В названиях css классов принято использовать снейк_кейс, а не камел_кейс

WebnautsAcademy commented on source/less/blocks/project.less:
```diff
@@ -0,0 +1,18 @@
+    margin-bottom: 40px;
+  }
+}
+.project__text .text {
```
Тут должны быть стили только одного блока. .text - ээто другой блок

WebnautsAcademy commented on source/less/variables.less:
```diff
@@ -2,16 +2,23 @@
 @bebasNeueRegular: "BebasNeueRegular";
 @proDisplayM: "SFProDisplayMedium";
-@proDisplayR: "SFProDisplay-Regular";
+@proDisplayR: "SFProDisplayRegular";
```
Шрифты заданы неверно. В имени шрифта должно быть только имя. Без жирности.

WebnautsAcademy reviewed (commented):
По поводу шрифтов, кажется, я уже писала. Почему не исправлено?

Шрифты надо подключать точно так же, как и в первом проекте.

ДЗ пока не принято. Поправьте моменты, что в комментариях.

Удачи! )

ThePen4iK committed:
remove mistakes but fonts don`t touch yet

С шрифтами пока трудности возникли, попытался сделать как Вы советуете, не получилось все шрифты слетели. Вернул назад, пока как было.
Я ж правильно понимаю что должно быть 2 переменных 1 для 1 шрифта и 2 для 2 шрифта. а поток в стилях просто прописывать fw 400 500 600...?
А подключать же надо в файле если взять как у меня называется setting.less как на 1 проекте ж?

WebnautsAcademy commented on source/index.html:
```diff
@@ -15,13 +15,19 @@
-            class="logo__img"
-          />
+          <picture>
+            <source
```
В source прописываются картинки с использованием медиа min-width/max-width. Это для адаптивности изображений: большие на десктопе, маленькие на мобильных и средние на планшете. И так же потом условие для webp картинок пропишется. Если пока Вы только десктоп делаете с ретиновыми картинками, то source не надо пока.

WebnautsAcademy commented on source/index.html:
```diff
@@ -15,13 +15,19 @@
+              media="(min-resolution: 1.5dppx)"
+              srcset="./img/logo@2x.png 2x"
+            />
+            <img
```
В теге img в теге pictute не проставляются width и height

WebnautsAcademy commented on source/index.html:
```diff
@@ -58,9 +64,12 @@ <h1 class="start__title title">
+                media="(min-resolution: 1.5dppx)"
+                srcset="./img/main@2x.png 2x"
+              />
               <img
```
Предыдущий вариант img был правильным. За исключением ширины и высоты. Они не пишутся, если img лежит в picture, как я выше написала

WebnautsAcademy commented on source/index.html:
```diff
@@ -83,38 +92,48 @@ <h1 class="start__title title">
         </div>
         <div class="about__wrapper-bottom">
-          <img
-            src="img/about.png"
```
И это было правильным, но без высоты и ширины... Проверьте все img. source только для адаптива и webp. Для ретины             srcset="./img/about@2x.png 2x". У Вас в некоторых местах 2x, что у закрывающей кавычки было утеряно. А именно это 2х и проверяет плотность пикселей экрана.

WebnautsAcademy commented on source/index.html:
```diff
@@ -271,69 +300,125 @@ <h4 class="gallery__info info">О проекте</h4>
+        <div class="review__wrapper">
+          <div class="review__item review__item-large">
+            <div class="review__wrapper-img">
+              <picture>
```
Тегам picture так же класс задается

WebnautsAcademy commented on source/less/blocks/about.less:
```diff
@@ -24,16 +23,8 @@
-  line-height: 22px;
-  text-align: center;
-
+  .circle;
```
. circle - не элемент about и не может тут стилизоваться. Если это элемент about, то его класс .about__circle

WebnautsAcademy commented on source/less/blocks/btn.less:
```diff
@@ -20,3 +20,9 @@
   }
 }
+.btn__view {
+  font-family: @bebasNeueRegular;
```
@bebasNeue

А шрифт должен подключаться в дерективе @font-face. Где в src указывается нужный файл в зависимости от жирности. Но имя семейства шрифтов - постоянно BebasNew.

WebnautsAcademy commented on source/less/setting.less:
```diff
@@ -1,3 +1,23 @@
+//   font-weight: 700;
+// }
+// @font-face {
+//   font-family: "BebasNeue";
```
Вообще они в файле fonts.less задаются, но если лекторы так сказали сделать - пусть будет. В src указывается реальный путь к файлу. Это никак с переменными лесс не связано. Тут просто путь к файлу со фрифтом нужной жирности. Пропишите его, как надо и шрифты перестанут отваливаться. 

В предыдущем проекте у Вас так шрифт гилрой подключался. Сделайте по аналогии. О чем я ранее и писала )

WebnautsAcademy reviewed (commented):
https://dev-gang.ru/article/adaptivnye-izobrazhenija-dlja-retina-displeev-na-html5-ggg23ic91r/ - тут всё, что о ретине и тегах img и picture Вам может пригодиться, чтобы мои правки правильно внести )

Удачи!

ThePen4iK committed:
fixed fonts picture and circle
